sqlmesh/core/scheduler.py

```diff
@@ -362,6 +362,10 @@ def start_date(
         elif start_dt:
             earliest = min(earliest, start_dt)
 
+    if earliest:
+        earliest = to_datetime(earliest)
```

Comment:
snapshot._start = to_datetime(earliest)

Reply:
Looks like earliest is already a datetime, so removing conversion